tests/unit/php/Core/Framework/DependencyInjection/CompilerPass/SalesChannelEntityCompilerPassTest.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Core\Framework\DependencyInjection\CompilerPass;
+
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\System\DependencyInjection\CompilerPass\SalesChannelEntityCompilerPass;
+use Shopware\Core\System\SalesChannel\Entity\SalesChannelRepository;
+use Symfony\Component\Config\FileLocator;use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+
+/**
+ * @internal
+ * @covers SalesChannelEntityCompilerPass
+ */
+class SalesChannelEntityCompilerPassTest extends TestCase
+{
+    public function testEntityRepositoryAutowiring(): void
+    {
+        $container = new ContainerBuilder();
+
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../../Resources/config'));
```

Comment:
Please just `$container->setDefinition` with just plain PHP so it is not depending on any loaders.

Reply:
No, i explained it jet.

Reply:
There no reason too use XML only when you are lazy. Just use plain PHP objects

Reply:
The reason are, we use XML by default. So we can it use by testing.
Else the other problems and fixes will not noticed by us?